File: swift/Sources/ErghiSDK/Config/ErghiConfig.swift
import Foundation

/// Configuration for Erghi SDK
public struct ErghiConfig: Sendable {
    /// Base API URL
    public let apiURL: URL
    
    /// WebSocket URL (optional, will be derived from apiURL if not provided)
    public let wsURL: URL?
    
    /// API Key (optional, can use JWT instead)
    public let apiKey: String?
    
    /// Client ID for M2M authentication
    public let clientId: String?
    
    /// Client Secret for M2M authentication
    public let clientSecret: String?
    
    /// Workspace ID
    public let workspaceId: String?
    
    /// Account ID
    public let accountId: String?
    
    /// Request timeout
    public let timeout: TimeInterval
    
    /// Enable debug logging
    public let debug: Bool
    
    public init(
        apiURL: URL,
        wsURL: URL? = nil,
        apiKey: String? = nil,
        clientId: String? = nil,
        clientSecret: String? = nil,
        workspaceId: String? = nil,
        accountId: String? = nil,
        timeout: TimeInterval = 30,
        debug: Bool = false
    ) {
        self.apiURL = apiURL
        self.wsURL = wsURL
        self.apiKey = apiKey
        self.clientId = clientId
        self.clientSecret = clientSecret
        self.workspaceId = workspaceId
        self.accountId = accountId
        self.timeout = timeout
        self.debug = debug
    }
    
    /// Get WebSocket URL (derive from apiURL if not provided)
    public var websocketURL: URL {
        if let wsURL = wsURL {
            return wsURL
        }
        
        var components = URLComponents(url: apiURL, resolvingAgainstBaseURL: false)!
        components.scheme = apiURL.scheme == "https" ? "wss" : "ws"
        return components.url!
    }
}

File: flutter/lib/src/config/erghi_config.dart
import 'package:equatable/equatable.dart';

/// Configuration for ErghiClient
class ErghiConfig extends Equatable {
  /// Base API URL
  final String apiUrl;

  /// WebSocket URL
  final String? wsUrl;

  /// API Key (optional, can use JWT instead)
  final String? apiKey;

  /// Client ID for M2M authentication
  final String? clientId;

  /// Client Secret for M2M authentication
  final String? clientSecret;

  /// Request timeout duration
  final Duration timeout;

  /// Enable debug logging
  final bool debug;

  const ErghiConfig({
    required this.apiUrl,
    this.wsUrl,
    this.apiKey,
    this.clientId,
    this.clientSecret,
    this.timeout = const Duration(seconds: 30),
    this.debug = false,
  });

  /// Get WebSocket URL (derive from apiUrl if not provided)
  String get websocketUrl {
    if (wsUrl != null) return wsUrl!;
    if (apiUrl.startsWith('https://')) {
      return apiUrl.replaceFirst('https://', 'wss://');
    }
    return apiUrl.replaceFirst('http://', 'ws://');
  }

  ErghiConfig copyWith({
    String? apiUrl,
    String? wsUrl,
    String? apiKey,
    String? clientId,
    String? clientSecret,
    Duration? timeout,
    bool? debug,
  }) {
    return ErghiConfig(
      apiUrl: apiUrl ?? this.apiUrl,
      wsUrl: wsUrl ?? this.wsUrl,
      apiKey: apiKey ?? this.apiKey,
      clientId: clientId ?? this.clientId,
      clientSecret: clientSecret ?? this.clientSecret,
      timeout: timeout ?? this.timeout,
      debug: debug ?? this.debug,
    );
  }

  @override
  List<Object?> get props => [apiUrl, wsUrl, apiKey, clientId, clientSecret, timeout, debug];
}
